dyn_sim/ctrl/segway/fdbk_lin_segway_ctrl.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+
+from dyn_sim.ctrl.ctrl import Controller
+from dyn_sim.sys.planar.segway import Segway
+
+# constants
+g = 9.80665  # gravitational acceleration
+
+
+class FLPosSegwayController(Controller):
+    """Feedback linearizing controller for segway.
+
+    Only capable of position regulation at the moment.
+    Could implement trajectory following using Callables.
+
+    Derivation of feedback linearizing controller available in
+    'CDS233: Project Submission 2'
+    """
+
+    def __init__(
+        self,
+        seg: Segway,
+        p_star: float,
+        alpha1: float,
+        alpha2: float,
+    ) -> None:
+        """Initialize a segway feedback linearizing controller.
+
+        Note by the Routh-Hurwitz criterion, for 2-d output dynamics,
+        a necessary and sufficient condition for (exponential) stability is both
+        alpha1 and alpha2 are positive.
+
+        Parameters
+        ----------
+        seg: Segway
+            Segway object to be controlled.
+        p_star: float
+            Position that segway will be regulated to.
+        alpha1: float
+            First gain for output dynamics.
+        alpha2: float
+            Second gain for output dynamics.
+        """
+        assert alpha1 > 0.0
+        assert alpha2 > 0.0
+
+        super(FLPosSegwayController, self).__init__(seg)
+
+        self._seg = seg
+        self._p_star = p_star
+        self._alpha1 = alpha1
+        self._alpha2 = alpha2
+
+    def ctrl(self, t: float, s: np.ndarray) -> np.ndarray:
+        """Feedback linearizing control law.
+
+        Parameters
+        ----------
+        t: float
+            Time.
+        s: np.ndarray
+            State.
+
+        Returns
+        -------
+        k: np.ndarray, shape = (1,)
```

Comment:
just call this u, if we ever used the convention k it would be for the function name, but that's already called ctrl.